src/hardware/driver/adc_internal.rs

```diff
@@ -0,0 +1,84 @@
+// This is a dummy driver for the Driver analog reads of the output Voltage and Current.
+// Exact Sacales and Pinout will be filled in once we have HW.
+
+use super::super::hal::{
+    adc,
+    gpio::{gpiof::*, Analog},
+    hal::blocking::delay::DelayUs,
+    prelude::*,
+    rcc::{rec, CoreClocks},
+    stm32::{ADC1, ADC2, ADC3},
+};
+
+const V_REF: f32 = 2.048; // ADC reference voltage
+const R_SENSE: f32 = 0.1; // Driver output current sense resistor (Will maybe be something else on HW)
+
+pub enum AdcChannel {
+    OutputVoltage(OutputChannelIdx),
+    OutputCurrent(OutputChannelIdx),
+}
+use super::OutputChannelIdx;
+
+pub struct AdcInternalPins {
+    pub output_voltage: (PF11<Analog>, PF3<Analog>), // JADC1_IN2_P, JADC3_IN5_P
+    pub output_current: (PF12<Analog>, PF4<Analog>), // JADC1_IN2_N, JADC3_IN5_N
+}
+
+pub struct AdcInternal {
+    adc1: adc::Adc<ADC1, adc::Enabled>,
+    adc3: adc::Adc<ADC3, adc::Enabled>,
+    pins: AdcInternalPins,
+}
+
+impl AdcInternal {
+    pub fn new(
+        delay: &mut impl DelayUs<u8>,
+        clocks: &CoreClocks,
+        adc_rcc: (rec::Adc12, rec::Adc3),
+        adc: (ADC1, ADC2, ADC3),
+        pins: AdcInternalPins,
+    ) -> Self {
+        // Setup ADCs
+        let (adc1, _adc2) = adc::adc12(adc.0, adc.1, delay, adc_rcc.0, clocks);
+        let adc3 = adc::Adc::adc3(adc.2, delay, adc_rcc.1, clocks);
+
+        let mut adc1 = adc1.enable();
+        adc1.set_resolution(adc::Resolution::SIXTEENBIT);
```

Comment:
Better explicitly set a sampling time for both ADCs now.
Even if #571 is unresolved.

Reply:
I set it to a reasonable sampling time of 64 cycles now. With a 100 MHz clock I still eyeballed a difference between 8 and 16 cycles, so 64 cycles at 50 MHz should be on the safe side.